### PR TITLE
[WIP] add php 7.4 to travis

### PR DESCRIPTION
**Pull request by oallain**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets |  License         | MIT

Just add PHP version 7.4 in Travis.

### CODE REVIEW

**Comment by oallain** [comment]
In travis logs we can see a lot of :
`Deprecated: Array and string offset access syntax with curly braces is deprecated in /home/travis/build/Sylius/Sylius/vendor/behat/transliterator/src/Behat/Transliterator/Transliterator.php on line 357`

This is fixed in https://github.com/Behat/Transliterator/pull/29 we just need a new release.
Or we can switch to symfony/string :smile:

**Comment by pamil** [comment]
I'm good with using Symfony String component, although it's still experimental as of Symfony 5.0. On the other hand, I wouldn't put much hope in Behat/Transliterator getting tagged.

**Comment by pamil** [comment]
Anyway, if it's just a deprecation, I'd address it in an another PR.

File: src/Sylius/Component/Product/Generator/SlugGenerator.php [hunk 1]
```diff
@@ -22,9 +22,8 @@ final class SlugGenerator implements SlugGeneratorInterface
      */
     public function generate(string $name): string
     {
-        // Manually replacing apostrophes since Transliterator started removing them at v1.2.
-        $name = str_replace('\'', '-', $name);
+        $unicodeString = (new AsciiSlugger())->slug($name);
```
Comment (pamil) [inline src/Sylius/Component/Product/Generator/SlugGenerator.php]: Could be an one-liner:

```suggestion
        return (new AsciiSlugger())->slug($name)->lower()->toString();
```

**Comment by oallain** [comment]
I don't understand why a got this error in Behat :(

**Comment by pamil** [comment]
Btw. `behat/transliterator` v1.3.0 has been released, so we don't need to change to `symfony/string` yet. I have no idea why it fails though :(

**Comment by oallain** [comment]
Good news for `behat/transliterator` i revert :smile:

**Comment by oallain** [comment]
I nned help :sos:

**Comment by diimpp** [comment]
Would be cool to have before 1.7 release :)

**Comment by lchrusciel** [comment]
Too late, but I suppose we can think about adding support for 7.4 in patch ;)

**Comment by pamil** [comment]
The base of this pull-request was changed, you need fetch and reset your local branch 
if you want to add new commits to this pull request. **Reset before you pull, else commits
may become messed-up.**

Unless you added new commits (to this branch) locally that you did not push yet, 
execute `git fetch origin && git reset "travis-php74"` to update your local branch.
 
Feel free to ask for assistance when you get stuck :+1:

**Comment by pamil** [comment]
Rebased to 1.7, let's see what would happen to the build 🎉

File: .travis.yml [hunk 2]
```diff
@@ -1,7 +1,5 @@
 language: php
 
-dist: trusty
```
Comment (Zales0123) [inline .travis.yml:3]: I'm struggling with javascript scenarios on some newer version of Travis ubuntu for some time already (see [this](https://github.com/Zales0123/Sylius/commit/1500de3d03143db3d7b57da914d953fdaa79581a)). I don't know why it does not work properly. We don't see it in the build, but the problem is `WebDriver\Exception\InvalidCookieDomain: invalid cookie domain`. Maybe you've already encountered it?

Nevertheless, for this PR, can't we stay with Ubuntu Trusty yet? Or PHP 7.4 is not available on it?

**Comment by pamil** [comment]
It's possible to use PHP 7.4 on Trusty, but there's no GD extension then.

**Comment by oallain** [comment]
Yes, You can see test at https://travis-ci.org/github/Sylius/Sylius/jobs/687353959

**Comment by pamil** [comment]
@oallain what about trying to get our current Travis configuration work with Xenial and then add PHP 7.4 once it passes?

**Comment by oallain** [comment]
> @oallain what about trying to get our current Travis configuration work with Xenial and then add PHP 7.4 once it passes?

Yep, we can probably do that but I don't have time this week :/

**Comment by pamil** [comment]
No worries! Tried to make it work in #11505 (going straight to Bionic 18.04 instead of Xenial 16.04), a few scenarios failing.

**Pull request closed without merging**